### PR TITLE
Wait for timestamps have been stubbed in UI tests

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What

- Using `textContent` instead of `innerHTML` because `innerHTML` is slow because HTML tags have to be searched and parsed into DOM nodes. 
- Wait for timestamps to stubbed

# Why

https://skype.visualstudio.com/SPOOL/_workitems/edit/2779986

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-composites/tests/browser/common/utils.ts [hunk 1]
```diff
@@ -238,7 +238,17 @@ export const waitForPiPiPToHaveLoaded = async (page: Page, videosEnabledCount: n
 export const stubMessageTimestamps = async (page: Page): Promise<void> => {
   const messageTimestampId: string = dataUiId(IDS.messageTimestamp);
   await page.evaluate((messageTimestampId) => {
-    Array.from(document.querySelectorAll(messageTimestampId)).forEach((i) => (i.innerHTML = 'timestamp'));
+    Array.from(document.querySelectorAll(messageTimestampId)).forEach((i) => (i.textContent = 'timestamp'));
+    (function checkForStubbedTimestamps(i) {
+      setTimeout(function () {
```
Comment (JamesBurnside) [inline packages/react-composites/tests/browser/common/utils.ts:243]: avoid our own setTimeouts, instead use `waitForFunction`

so something like:
```ts
// Set timestamps in the DOM
await page.evaluate((messageTimestampId) => {
  Array.from(document.querySelectorAll(messageTimestampId)).forEach((i) => (i.textContent = 'timestamp'));
}, messageTimestampId);

// Wait for timestamps to have been updated in the DOM
// eslint-disable-next-line @typescript-eslint/no-explicit-any
await waitForFunction(page, (args: any) => {
  const timestampNodes = document.querySelectorAll(args.messageTimestampId);
  return timestampNodes.every(node => node.innerHTML === 'timestamp');
}, {
  messageTimestampId: messageTimestampId
})
```
(^disclaimer: completely untested code written in the github browser comment window, likely has syntax and logical errors)

Using waitForFunction both allows avoiding custom timeouts and uses the global timeout system, but is also wrapped in a takeScreenshotOnFailure in the event of a timeout happening.

Comment (JamesBurnside) [reply packages/react-composites/tests/browser/common/utils.ts:243]: waitForFunction will also fail the test here if the timestamps never updated

Comment (mgamis-msft) [reply packages/react-composites/tests/browser/common/utils.ts:243]: Thanks for the code snippet. It worked for me. Before I commit this, I was wondering does `waitForFunction` retry when `timestampNodes.every(node => node.innerHTML === 'timestamp')` returns false?

Comment (JamesBurnside) [reply packages/react-composites/tests/browser/common/utils.ts:243]: It should do yep. It should poll every x seconds (or every animationFrameRequest depending on what the default polling type is) and then has an overall timeout: https://playwright.dev/docs/api/class-page#page-wait-for-function (which we limit to 5 seconds)
image

**Review by JamesBurnside — APPROVED** [review]

**Pull request merged**